core/bazooka.go

```diff
@@ -337,6 +337,90 @@ func (b *Bazooka) DecodeState(stateBytes []byte) (ID, balance, nonce, token *big
 // Transactions
 //
 
+// SubmitBatch submits the batch on chain with updated root and compressed transactions
+func (b *Bazooka) SubmitBatch(commitments []Commitment) error {
+	b.log.Info(
+		"Attempting to submit a new batch",
+		"NumOfCommitments",
+		len(commitments),
+	)
+
+	if len(commitments) == 0 {
+		b.log.Info("No transactions to submit, waiting....")
+		return nil
+	}
+
+	var txs [][]byte
+	var updatedRoots [][32]byte
+	var aggregatedSig [][2]*big.Int
+	var totalTxs int
+
+	for _, commitment := range commitments {
+		compressedTxs, err := b.CompressTxs(commitment.Txs)
+		if err != nil {
+			b.log.Error("Unable to compress txs", "error", err)
+			return err
+		}
+		txs = append(txs, compressedTxs)
+		updatedRoots = append(updatedRoots, commitment.UpdatedRoot)
+		totalTxs += len(commitment.Txs)
+
+		// TODO cleanup
```

Comment:
Do we want to create an issue that adds an AggregatedSignature wrapper with a method like `ToSol()`?

Reply:
https://github.com/thehubbleproject/hubble-commander/issues/56